mllib/src/main/scala/org/apache/spark/ml/feature/Word2Vec.scala

```diff
@@ -194,6 +232,285 @@ final class Word2Vec @Since("1.4.0") (
 
   @Since("1.4.1")
   override def copy(extra: ParamMap): Word2Vec = defaultCopy(extra)
+
+  /**
+   * Similar to InitUnigramTable in the original code. Instead of using an array of size 100 million
+   * like the original, we size it to be 20 times the vocabulary size.
+   * We sacrifice memory here, to get constant time lookups into this array when generating
+   * negative samples.
+   */
+  private def generateUnigramTable(normalizedWeights: Array[Double], tableSize: Int): Array[Int] = {
+    val table = Array.fill(tableSize)(0)
+    var a = 0
+    var i = 0
```

Comment:
Could you please use more descriptive variable names here? I was expecting ``i`` to be the index into the table array